errors/error.go

```diff
@@ -287,7 +287,9 @@ func (e *Error) error(fields []interface{}, verbose bool) string {
 							v.End.Line, v.End.Column, v.End.Byte),
 					)
 				} else {
-					errParts = append(errParts, v.String())
+					copiedRange := v
+					copiedRange.Filename = filename(copiedRange.Filename)
+					errParts = append(errParts, copiedRange.String())
```

Comment:
`a.b = anyfunc(a.b)` only makes sense if it modify its input.

Reply:
I disagree, something like this:

```
a = copy(b)
```

Could create a deep and exact copy of b and return it. And then for filename I thought it was just some sort of constructor since I read `filename(v.Filename)`.

Reply:
Anothjer source of confusion is the name copiedRange, it made me think that the whole point was to create an exact copy of the other range, and then the filename function was just part of the copying process (some sort of deep copy), and then it prompted my question.

Reply:
the copy example just proves my point, `a.b` after the funcall is different (ie points to a different place or has its value changed somehow) otherwise it's a no-op.

Reply:
the `filename()` returns a filename suitable to be part of the error. It's inside the error package.

Reply:
I change to `cleanFilename()` but it's not a good name because it returns a different filename if it's a generated expression...

Reply:
> the copy example just proves my point, `a.b` after the funcall is different (ie points to a different place or has its value changed somehow) otherwise it's a no-op.

I don't get it how it makes your point =P